melusine/models/train.py

```diff
@@ -365,12 +368,54 @@ def predict_proba(self, X, **kwargs):
         Parameters
         ----------
         X : pd.DataFrame
-
+        prediction_interval : float, optional
+            between [0,1], the confidence level of the interval.
+            Only available with tensorflow-probability models.
         Returns
         -------
-        np.array
+        score : np.array
+            The estimation of probability for each category.
+        inf : np.array, optional
+            The upper bound of the estimation of probability.
+            Only provided if `prediction_interval` exists.
+        sup : np.array, optional
+            The lower bound of the estimation of probability.
+            Only provided if `prediction_interval` exists.
         """
 
+        X_input = self.prepare_email_to_predict(X)
+        if self.model.layers[-1].get_config().get('convert_to_tensor_fn') == 'mode':
```

Comment:
this case is true for probabilistic models :  
we use the distributions (which are the outputs of the model) and compute mean/std to estimate lower/upper bounds of uncertainty